File: android/src/main/kotlin/com/litedevs/is_tv/IsTVPlugin.kt
package com.litedevs.is_tv

import android.content.pm.PackageManager
import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** IsTVPlugin */
class IsTVPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  private var pm : PackageManager? = null

  private val isTV : Boolean by lazy {
    pm?.hasSystemFeature(PackageManager.FEATURE_LEANBACK) ?: false
  }

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "is_tv")
    channel.setMethodCallHandler(this)

    pm = flutterPluginBinding.applicationContext.packageManager;
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "check") {
      result.success(isTV)
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
    pm = null;
  }
}

File: macos/Classes/IsTVPlugin.swift
import Cocoa
import FlutterMacOS

public class IsTVPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "is_tv", binaryMessenger: registrar.messenger)
    let instance = IsTVPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
      if(call.method == "check") {
        return result(false);
      result(FlutterMethodNotImplemented)
    }
  }
}
